tests/kafkatest/tests/core/delegation_token_test.py

```diff
@@ -0,0 +1,128 @@
+
+        self.producer = VerifiableProducer(self.test_context, num_nodes=1, kafka=self.kafka, topic=self.topic, max_messages=1,
+                                       throughput=1, kafka_opts_override=self.client_kafka_opts,
+                                       client_prop_file_override=self.client_properties_content)
+
+        self.consumer = ConsoleConsumer(self.test_context, num_nodes=1, kafka=self.kafka, topic=self.topic,
+                                        kafka_opts_override=self.client_kafka_opts,
+                                        client_prop_file_override=self.client_properties_content,
+                                        consumer_properties={"client.id":"console-consumer"})
+
+    def setUp(self):
+        self.zk.start()
+
+    def tearDown(self):
+        self.producer.nodes[0].account.remove(self.jaas_deleg_conf_path)
+        self.consumer.nodes[0].account.remove(self.jaas_deleg_conf_path)
+
+    def generate_delegation_token(self):
+        self.logger.debug("Request delegation token")
+        self.delegation_tokens.generate_delegation_token()
+        self.jaas_deleg_conf = self.delegation_tokens.create_jaas_conf_with_delegation_token()
+
+    def expire_delegation_token(self):
+        self.kafka.client_sasl_mechanism = 'GSSAPI,SCRAM-SHA-256'
+        token_hmac = self.delegation_tokens.token_hmac()
+        self.delegation_tokens.expire_delegation_token(token_hmac)
+
+
+    def produce_with_delegation_token(self):
+        self.producer.acked_values = []
+        self.producer.nodes[0].account.create_file(self.jaas_deleg_conf_path, self.jaas_deleg_conf)
+        self.logger.debug(self.jaas_deleg_conf)
+        self.producer.start()
+
+    def consume_with_delegation_token(self):
+        self.logger.debug("Consume messages with delegation token")
+
+        self.consumer.nodes[0].account.create_file(self.jaas_deleg_conf_path, self.jaas_deleg_conf)
+        self.logger.debug(self.jaas_deleg_conf)
+        self.consumer.consumer_timeout_ms = 5000
+
+        self.consumer.start()
+        self.consumer.wait()
+
+    def get_datetime_ms(self, input_date):
+        return int(time.mktime(datetime.strptime(input_date,"%Y-%m-%dT%H:%M").timetuple()) * 1000)
+
+    def renew_delegation_token(self):
+        dt = self.delegation_tokens.parse_delegation_token_out()
+        orig_expiry_date_ms = self.get_datetime_ms(dt["expirydate"])
+        new_expirydate_ms = orig_expiry_date_ms + 1000
+
+        self.delegation_tokens.renew_delegation_token(dt["hmac"], new_expirydate_ms)
+
+    def test_delegation_token_lifecycle(self, security_protocol='SASL_PLAINTEXT', sasl_mechanism='GSSAPI'):
+        self.kafka.security_protocol = security_protocol
+        self.kafka.client_sasl_mechanism = 'GSSAPI,SCRAM-SHA-256'
+        self.kafka.interbroker_sasl_mechanism = sasl_mechanism
+        self.kafka.start()
+
+        self.generate_delegation_token()
```

Comment:
As I understand you validate with producing and consuming after renewal. Do you think it would make sense to do it after this step too?